app/services/mixtape_generator.rb

```diff
@@ -0,0 +1,20 @@
+class MixtapeGenerator
+  def initialize(genre:)
+    @genre = genre
+  end
+
+  def self.perform(genre:)
+    new(genre: genre).perform
+  end
+
+  def perform
+    Mixtape.new(genre: @genre, songs: pick_songs)
+  end
+
+  def pick_songs
+    Song.order_by_rand
+        .includes(:genre, :artist)
+        .joins(:genre, :artist)
+        .where("lower(genres.name) = ?", @genre.downcase)
```

Comment:
So, why not find the `Genre` by its id and do `songs.genre_id = ?. genre.id`?

Ahhhhhh... see I got confused by seeing `genre` and I thought the entity was being passed it, but it isn't. It is `genre_name` as the parameter.

Maybe change the `def self.perform(genre:)` to `def self.perform(genre_name:)` to clarify.

If I see the named argument, `genre` I'd think to pass a fully hydrated `Genre` item.